webapp/src/main/java/com/box/l10n/mojito/service/branch/notification/BranchNotificationMessageSendersConfigurationProperties.java

```diff
@@ -31,4 +41,104 @@ public void setEnabled(boolean enabled) {
       this.enabled = enabled;
     }
   }
+
+  static class SlackConfigurationProperties {
+
+    String slackClientId;
+
+    MessageBuilderConfigurationProperties messages;
+
+    String userEmailPattern;
+
+    boolean useDirectMessage = false;
+
+    public String getSlackClientId() {
+      return slackClientId;
+    }
+
+    public void setSlackClientId(String slackClientId) {
+      this.slackClientId = slackClientId;
+    }
+
+    public MessageBuilderConfigurationProperties getMessages() {
+      return messages;
+    }
+
+    public void setMessages(MessageBuilderConfigurationProperties messages) {
+      this.messages = messages;
+    }
+
+    public String getUserEmailPattern() {
+      return userEmailPattern;
+    }
+
+    public void setUserEmailPattern(String userEmailPattern) {
+      this.userEmailPattern = userEmailPattern;
+    }
+
+    public boolean isUseDirectMessage() {
+      return useDirectMessage;
+    }
+
+    public void setUseDirectMessage(boolean useDirectMessage) {
+      this.useDirectMessage = useDirectMessage;
+    }
+
+    public static class MessageBuilderConfigurationProperties {
+
+      String newStrings =
```

Comment:
@maallen I tried to refactor names so that they are consistent. We'll need to be careful when updating the configuration to use the new names